types/eslint-utils/index.d.ts

```diff
@@ -0,0 +1,284 @@
+// Type definitions for eslint-utils 3.0
+// Project: https://github.com/mysticatea/eslint-utils#readme
+// Definitions by: Josh Goldberg <https://github.com/JoshuaKGoldberg>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+import { AST, Scope, SourceCode } from 'eslint';
+import { VisitorKeys } from 'eslint-visitor-keys';
+import { Comment, Node, SourceLocation } from 'estree';
+
+type CommentOrToken = Comment | AST.Token;
+
+/**
+ * Find the variable of a given name.
+ * @param initialScope The scope to start finding.
+ * @param nameOrNode The variable name to find. If this is a Node object then it should be an Identifier node.
+ * @returns The found variable or null.
+ */
+export function findVariable(initialScope: Scope.Scope, nameOrNode: string | Node): Scope.Variable | null;
+
+/**
+ * Get the location of the given function node for reporting.
+ * @param node The function node to get.
+ * @param sourceCode The source code object to get tokens.
+ * @returns The location of the function node for reporting.
+ */
+export function getFunctionHeadLocation(node: Node, sourceCode: SourceCode): SourceLocation;
+
+/**
+ * Get the name and kind of the given function node.
+ * @param node The function node to get.
+ * @param sourceCode The source code object to get the code of computed property keys.
+ * @returns The name and kind of the function node.
+ */
+export function getFunctionNameWithKind(node: Node, sourceCode?: SourceCode): string;
+
+/**
+ * Get the innermost scope which contains a given location.
+ * @param initialScope The initial scope to search.
+ * @param node The location to search.
+ * @returns The innermost scope.
+ */
+export function getInnermostScope(initialScope: Scope.Scope, node: Node): Scope.Scope;
+
+/**
+ * Get the property name from a MemberExpression node or a Property node.
+ * @param node The node to get.
+ * @param initialScope The scope to start finding variable.
+ * @returns The property name of the node.
+ * @remarks
+ * If the node is a computed property node and a scope was given, this checks the computed property name by the `getStringIfConstant` function with the scope, and returns the value of it.
+ */
+export function getPropertyName(node: Node, initialScope?: Scope.Scope): string | null;
+
+export type StaticValue = StaticValueProvided | StaticValueOptional;
+
+export interface StaticValueProvided {
+    value: any;
```

Comment:
is there a better type we can use for this?

Reply:
🤔, I was just going off of https://eslint-utils.mysticatea.dev/api/ast-utils.html#getstaticvalue for the `any` and wanted to play it safe. Looking through https://github.com/JoshuaKGoldberg/eslint-utils/blob/99fd4befbea3176efc165e9dd9f6dd62cb57feff/test/get-static-value.js it looks like it's what it looks like: the inferable primitive static value types (`string | number | ...`). 👍

Reply:
🤔, I was just going off of https://eslint-utils.mysticatea.dev/api/ast-utils.html#getstaticvalue for the any and wanted to play it safe. Looking through https://github.com/mysticatea/eslint-utils/blob/99fd4befbea3176efc165e9dd9f6dd62cb57feff/test/get-static-value.js it can be a bunch of different strange values:

```ts
{ code: "Symbol.iterator", expected: { value: Symbol.iterator } },
...
{ code: "Object.freeze", expected: { value: Object.freeze } },
```

...so `any` feels appropriate here to me.

Reply:
Or mark it as generic type `T` and defaults to be `unknown`?